api/python/quilt3/data_transfer.py

```diff
@@ -151,10 +151,8 @@ def find_correct_client(self, api_type, bucket, param_dict):
                 else:
                     raise S3NoValidClientError(f"S3 AccessDenied for {api_type} on bucket: {bucket}")
 
-    def get_boto_session(self):
-        botocore_session = create_botocore_session()
-        boto_session = boto3.Session(botocore_session=botocore_session)
-        return boto_session
+    def get_boto_session(self) -> boto3.Session:
+        return get_boto3_session() or boto3.Session()
```

Comment:
@nl0 
it feels like it's not right to have fallback logic here
I don't quite like adding `fallback` parameter to `get_boto3_session()` because it would have `Optional[]` as return type and you have to check `assert session is not None` even if `fallback=True` (though maybe some type annotations magic exists to handle that)
maybe it's not useful to have `get_boto3_session()` returning `None` after all? 🤔